overlord/snapstate/component_install_test.go

```diff
@@ -167,22 +178,29 @@ version: 1.0
 }
 
 func createTestSnapInfoForComponent(c *C, snapName string, snapRev snap.Revision, compName string) *snap.Info {
-	return createTestSnapInfoForComponentWithType(c, snapName, snapRev, compName, "test")
+	return createTestSnapInfoForComponents(c, snapName, snapRev, map[string]string{compName: "test"})
 }
 
-func createTestSnapInfoForComponentWithType(c *C, snapName string, snapRev snap.Revision, compName, compType string) *snap.Info {
+func createTestSnapInfoForComponents(c *C, snapName string, snapRev snap.Revision, compNamesToType map[string]string) *snap.Info {
 	snapType := "app"
-	if compType == "kernel-modules" {
-		snapType = "kernel"
+	for _, typ := range compNamesToType {
+		if typ == "kernel-modules" {
+			snapType = "kernel"
+		}
 	}
-	snapYaml := fmt.Sprintf(`name: %s
+
+	var b bytes.Buffer
+	fmt.Fprintf(&b, `name: %s
 type: %s
 version: 1.1
 components:
-  %s:
-    type: %s
-`, snapName, snapType, compName, compType)
-	info, err := snap.InfoFromSnapYaml([]byte(snapYaml))
+`, snapName, snapType)
+
+	for compName, typ := range compNamesToType {
+		fmt.Fprintf(&b, "  %s:\n    type: %s\n", compName, typ)
+	}
```

Comment:
is this left over debug code?

Reply:
This is a `Fprintf` writing to `b`, not a regular `Printf`. It is building a snap.yaml that includes a dynamic set of components.